### PR TITLE
test: [M3-7053] - Add Cypress integration test for VPC create flow

### PR DESCRIPTION
**Pull request by jdamore-linode**
## Description 📝
Adds a Cypress integration test for the VPC create flow.

## Major Changes 🔄
```
   - Confirms VPC creation flow using mock API data.
   - Confirms that users can add and remove subnets during create flow.
   - Confirms client side validation when entering invalid labels and IP ranges.
   - Confirms that UI handles API errors gracefully.
   - Confirms that UI redirects to created VPC page after creating a VPC.
```

## How to test 🧪
We can refer to the automated test run, but to run the tests locally you can use this command:

```
yarn cy:run -s "cypress/e2e/core/vpc/vpc-create.spec.ts"
```

### CODE REVIEW

**Review by hana-linode — APPROVED** [review]
Ran the tests locally and they passed 🎉

File: packages/manager/cypress/e2e/core/vpc/vpc-create.spec.ts [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+    cy.visitWithLogin('/vpcs/create');
+    cy.wait(['@getFeatureFlags', '@getClientstream']);
+
+    cy.findByText('Region')
+      .should('be.visible')
+      .click()
+      .type(`${vpcRegion.label}{enter}`);
+
+    cy.findByText('VPC label').should('be.visible').click().type(mockVpc.label);
```
[the remark this answers is not in the document]
Comment (coliu-akamai) [reply packages/manager/cypress/e2e/core/vpc/vpc-create.spec.ts:90]: +1 makes sense to me too -- I'll ask UX to double check :D

**Review by coliu-akamai — APPROVED** [review]
✅ confirmed tests pass. Thanks Joe!

File: packages/manager/cypress/e2e/core/vpc/vpc-create.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (coliu-akamai) [reply packages/manager/cypress/e2e/core/vpc/vpc-create.spec.ts:90]: confirmed that label should be capitalized :D

[the remark this answers is not in the document]
Comment (jdamore-linode) [reply packages/manager/cypress/e2e/core/vpc/vpc-create.spec.ts:90]: Thanks for confirming @coliu-akamai! I'll make those changes in this PR if that works for everyone 👍

File: packages/manager/cypress/e2e/core/vpc/vpc-create.spec.ts [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+      description: randomPhrase(),
+      subnets: mockSubnets,
+    });
+
+    const ipValidationErrorMessage = 'The IPv4 range must be in CIDR format';
+    const vpcCreationErrorMessage = 'An unknown error has occurred.';
+    const totalSubnetUniqueLinodes = mockSubnets.reduce(
+      (acc: number, cur: Subnet) => acc + cur.linodes.length,
+      0
+    );
```
[the remark this answers is not in the document]
Comment (jdamore-linode) [reply packages/manager/cypress/e2e/core/vpc/vpc-create.spec.ts:75]: Thanks @hana-linode! I'll take a look at that and see if I can use it from the test

File: packages/manager/cypress/e2e/core/vpc/vpc-create.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (coliu-akamai) [reply packages/manager/cypress/e2e/core/vpc/vpc-create.spec.ts:90]: thank you!

**Comment by coliu-akamai** [comment]
Hey @jdamore-linode! Just wanted to give a heads-up that the VPC Create flow will be changed a from this PR/ticket:
https://github.com/linode/manager/pull/9751

The changes are:
- All subnets have the remove [x] button, including the first subnet
- The button text for adding subnets is changed from “Add Subnet” to “Add Another Subnet”
  -  If there are no subnets, it's "Add a Subnet"
- There's a divider below each subnet now

**Comment by jdamore-linode** [comment]
> Hey @jdamore-linode! Just wanted to give a heads-up that the VPC Create flow will be changed a from this PR/ticket: #9751
> 
> The changes are:
> 
> * All subnets have the remove [x] button, including the first subnet
> * The button text for adding subnets is changed from “Add Subnet” to “Add Another Subnet” (affects line 156, 199 most likely)
>   
>   * If there are no subnets, it's "Add a Subnet"
> * There's a divider below each subnet now

Thanks @coliu-akamai and @hana-linode for the heads up! I'll review #9751 later today, and my plan is to wait for that work to be reviewed and merged, and then I'll circle back here and update this PR to account for these changes and add additional coverage where necessary (being able to delete the first subnet specifically comes to mind)

**Comment by jdamore-linode** [comment]
@coliu-akamai @hana-linode Just letting you know I pushed some changes to account for the recent improvements made to the VPC create page:

- Updated tests to account for "Add a Subnet" label change to "Add another Subnet"
- Added a new test to create a VPC without Subnets (confirms user can delete the subnet on the create page, confirms that button label changes to "Add a Subnet", etc.)

I don't know if this warrants a re-review but wanted to give you both a heads up in case you wanted to put some eyes on it!

File: packages/manager/src/features/VPCs/VPCCreate/SubnetNode.tsx [hunk 3]
```diff
@@ -63,15 +63,15 @@ export const SubnetNode = (props: Props) => {
             disabled={disabled}
             errorText={subnet.labelError}
             inputId={`subnet-label-${idx}`}
-            label="Subnet label"
+            label="Subnet Label"
```
Comment (coliu-akamai) [inline packages/manager/src/features/VPCs/VPCCreate/SubnetNode.tsx:66]: sorry I forgot to mention -- this change will affect a bunch of tests in the `VPCs/VPCCreate` folder and the `SubnetCreateDrawer.test.tsx` file too. It should be a super quick fix (just updating everything to 'Subnet Label' in those tests, but lmk if you've questions!

**Review by coliu-akamai — APPROVED** [review]
Other than missing a new Tests changeset and the minor issue I commented about, this looks good + the cypress tests passed. Reapproving on those being resolved -- thanks Joe!!

File: packages/manager/src/features/VPCs/VPCCreate/SubnetNode.tsx (hunk 3 above)
Comment (jdamore-linode) [reply packages/manager/src/features/VPCs/VPCCreate/SubnetNode.tsx:66]: > sorry I forgot to mention -- this change will affect a bunch of tests in the `VPCs/VPCCreate` folder and the `SubnetCreateDrawer.test.tsx` file too. It should be a super quick fix (just updating everything to 'Subnet Label' in those tests, but lmk if you've questions!

Ah, thank you for the heads up! I'll take care of that soon. Shame on me for not considering the unit tests 😅

Comment (coliu-akamai) [reply packages/manager/src/features/VPCs/VPCCreate/SubnetNode.tsx:66]: awesome thanks!

**Comment by coliu-akamai** [comment]
@jdamore-linode  Oh one quick thing, I saw that you changed the Subnet delete dialog to be 'Subnet Label' instead of 'Subnet label' too -- this will affect some of the cypress tests too 😅 (vpc-details-page.spec.ts, line 174, line 241 I think)

**Comment by jdamore-linode** [comment]
> @jdamore-linode Oh one quick thing, I saw that you changed the Subnet delete dialog to be 'Subnet Label' instead of 'Subnet label' too -- this will affect some of the cypress tests too 😅 (vpc-details-page.spec.ts, line 174, line 241 I think)

Classic. 😅 Thanks Connie!

**Comment by jdamore-linode** [comment]
Going to merge this despite the E2E failure. The failing tests are caused by a gateway timeout when attempting to create LKE clusters, so it's not related to the work in this PR.

**Pull request merged**